workout/views.py

```diff
@@ -1,39 +1,97 @@
-from django.shortcuts import render
+from django.shortcuts import render, get_object_or_404
 from django.views import generic
 from django.contrib.auth import authenticate, login
 from django.urls import reverse
 from django.http import HttpResponseRedirect, HttpResponse
 from .forms import WorkoutForm
 from workout.models import Workout
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from userprofile.forms import UserProfileForm
+from userprofile.serializers import UserProfileSerializer
+from rest_framework.renderers import TemplateHTMLRenderer
 
-class WorkoutView(generic.TemplateView):
+class WorkoutView(LoginRequiredMixin,generic.TemplateView):
     template_name = 'workout/workout.html'
+    login_url = '/login/'
```

Comment:
@regilkent123 @iamkatrinaanne you guys can just set `LOGIN_URL` in the settings.py file instead of adding `login_url` for every View with login required mixin